python/cudf/cudf/core/column/categorical.py

```diff
@@ -783,9 +785,9 @@ def to_pandas(
         nullable: bool = False,
         arrow_type: bool = False,
     ) -> pd.Index:
-        if nullable:
+        if arrow_type or self.dtype_enum in {2}:
```

Comment:
We would only want `self.dtype_enum` to influence `to_pandas` in pandas compatible mode, correct? Otherwise, a non cudf.pandas user who calls e.g. `to_pandas(arrow_type=False)` may still get an Arrow dtype pandas object back